otx/algorithms/segmentation/adapters/mmseg/configurer.py

```diff
@@ -0,0 +1,620 @@
+                        "unlabeled",
+                        "train_dataloader",
+                        "val_dataloader",
+                        "test_dataloader",
+                        "unlabeled_dataloader",
+                    ]
+                }
+            )
+
+            for subset in ["train", "val", "test", "unlabeled"]:
+                if subset not in cfg.data:
+                    continue
+                dataloader_cfg = cfg.data.get(f"{subset}_dataloader", None)
+                if dataloader_cfg is None:
+                    raise AttributeError(f"{subset}_dataloader is not found in config.")
+                dataloader_cfg = {**global_dataloader_cfg, **dataloader_cfg}
+                cfg.data[f"{subset}_dataloader"] = dataloader_cfg
+
+        _configure_dataloader(cfg)
+
+
+class IncrSegmentationConfigurer(SegmentationConfigurer):
+    """Patch config to support incremental learning for semantic segmentation."""
+
+    def configure_task(self, cfg, training):
+        """Patch config to support incremental learning."""
+        super().configure_task(cfg, training)
+
+        new_classes = np.setdiff1d(self.model_classes, self.org_model_classes).tolist()
+
+        # Check if new classes are added
+        has_new_class = len(new_classes) > 0
+
+        # Update TaskAdaptHook (use incremental sampler)
+        task_adapt_hook = ConfigDict(
+            type="TaskAdaptHook",
+            src_classes=self.org_model_classes,
+            dst_classes=self.model_classes,
+            model_type=cfg.model.type,
+            sampler_flag=has_new_class,
+            efficient_mode=cfg["task_adapt"].get("efficient_mode", False),
+        )
+        update_or_add_custom_hook(cfg, task_adapt_hook)
+
+
+class SemiSLSegmentationConfigurer(SegmentationConfigurer):
+    """Patch config to support semi supervised learning for semantic segmentation."""
+
+    def configure_data(self, cfg, training, data_cfg):
+        """Patch cfg.data."""
+        super().configure_data(cfg, training, data_cfg)
+        # Set unlabeled data hook
+        if training:
+            if cfg.data.get("unlabeled", False) and cfg.data.unlabeled.get("otx_dataset", False):
+                self.configure_unlabeled_dataloader(cfg)
+
+    def configure_task(self, cfg, training, **kwargs):
+        """Adjust settings for task adaptation."""
+        super().configure_task(cfg, training, **kwargs)
+
+        # Don't pass task_adapt arg to semi-segmentor
+        if cfg.model.type != "ClassIncrEncoderDecoder" and cfg.model.get("task_adapt", False):
+            cfg.model.pop("task_adapt")
+
+        # Remove task adapt hook (set default torch random sampler)
+        remove_custom_hook(cfg, "TaskAdaptHook")
+
+    @staticmethod
+    def configure_unlabeled_dataloader(cfg: Config):
+        """Patch for unlabled dataloader."""
+
+        model_task = {"classification": "mmcls", "segmentation": "mmdet", "segmentation": "mmseg"}  # noqa
```

Comment:
If hard to remove, try to import dataset and dataloader builders from mmseg directly.

Reply:
Thanks for good catch, but this PR couldn't remove `training_base.py` and `nncf_base.py` due to remained tasks (action). How about to remove task agnostic codes in other PR? Classification, detection also have same code.

Reply:
We do not fully agree on concept of global configurer, so there is many duplicated codes. I think we should disassemble configurer in the end, and I think it could be done in next refactoring phase. This refactoring phase's goal is collecting inner patches in one place so we can discuss how we can reduce inner patches

Reply:
It doesn't seem right that mmsegtask class mentions other task such mmcls and mmdet as of now. But I will approve this believing that this module is moved to where global configs are in the near future.

Reply:
Thanks for understanding, we will discuss about it next developer meeting